### PR TITLE
Breaking change: migrate to Pydantic 2.x

### PR DESCRIPTION
**Pull request by Mark90**
Breaking changes:
* Removed `class UniqueConstrainedList` -> use `unique_conlist()`, `Annotated[list, AfterValidator(validate_unique_items)]` or  `Annotated[set, ...]` instead
* Removed `class ChoiceList` -> use `choice_list()` instead
* Removed `class ContactPersonList` -> use `contact_person_list()` instead
* Moved `ReadOnlyField` from `pydantic_forms.core` to `pydantic_forms.validators`
* Changed `ReadOnlyField` from a pydantic Field to an Annotated Literal -> change your code from `field: int = ReadOnlyField(123)` to `field: ReadOnlyField(123)`

Tasks:
- [x] Self-review Mark
- [x] Self-review Maurits
- [x] Document all breaking changes and solutions

### CODE REVIEW

File: pydantic_forms/validators/components/accept.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Any, ClassVar, Optional
+
+from pydantic import GetCoreSchemaHandler, GetJsonSchemaHandler
+from pydantic.v1.errors import EnumMemberError
```
Comment (Mark90) [inline pydantic_forms/validators/components/accept.py:16]: Fix

File: pydantic_forms/validators/components/read_only_field.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    type_ = type(default)
+    try:
+        return JSON_SCHEMA_TYPES[type_]
+    except KeyError:
+        types = ", ".join(t.__name__ for t in chain(JSON_SCHEMA_TYPES, STRINGY_TYPES))
+        raise TypeError(f"Cannot make a ReadOnlyField for type {type_}. Supported types: {types}")
+
+
+def _read_only_list(default: Any, default_type: Any | None, json_schema: dict) -> Any:
+    # TODO change to a standalone generic type, i.e. read_only_list[str](["foo", "bar"])
```
Comment (Mark90) [inline pydantic_forms/validators/components/read_only_field.py:44]: Fix

File: pydantic_forms/validators/components/read_only_field.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    return Annotated[
+        default_type,
+        Field(default, json_schema_extra=json_schema | {"const": default}),
+        AfterValidator(validate_list),
+        PlainSerializer(serialize_list, when_used="json"),
+    ]
+
+
+def ReadOnlyField(default: Any, default_type: Any | None = None) -> Any:
+    # TODO Change to read_only_field
```
Comment (Mark90) [inline pydantic_forms/validators/components/read_only_field.py:69]: Fix

File: pydantic_forms/validators/components/accept.py (hunk 1 above)
Comment (Mark90) [reply pydantic_forms/validators/components/accept.py:16]: done

File: pydantic_forms/validators/components/read_only_field.py (hunk 2 above)
Comment (Mark90) [reply pydantic_forms/validators/components/read_only_field.py:44]: Created issue #16

**Review by mrijk — APPROVED** [review]
LGTM

**Pull request merged**